groups/nts/ntsa/ntsa_zerocopy.h

```diff
@@ -0,0 +1,273 @@
+// Copyright 2020-2023 Bloomberg Finance L.P.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef INCLUDED_NTSA_ZEROCOPY
+#define INCLUDED_NTSA_ZEROCOPY
+
+#include <bsls_ident.h>
+BSLS_IDENT("$Id: $")
+
+#include <ntscfg_platform.h>
+#include <ntsscm_version.h>
+
+#include <bslh_hash.h>
+#include <bsl_ostream.h>
+
+namespace BloombergLP {
+namespace ntsa {
+
+/// Provide a type holding a result of send() with copy avoidance mechanis.
+///
+/// @details
+/// Provide a value-semantic type that holds a result of send() with
+/// MSG_ZEROCOPY flag set.
+///
+/// @par Attributes
+/// This class is composed of the following attributes:
+///
+/// @li @b d_from:
+/// Number identifying the start of the range for which this result relates
+/// (inclusive)
+///
+/// @li @b d_to:
+/// Number identifying the end of the range for which this result relates
+/// (inclusive)
+///
+/// @li @b d_code:
+/// Code describing result of copy avoidance mechanism application.
+///
+/// @par Thread Safety
+/// This class is not thread safe.
+class ZeroCopy
+{
+    bsl::uint32_t d_from;
+    bsl::uint32_t d_to;
+    bsl::uint8_t  d_code;
```

Comment:
I have doubts if I should introduce a separate enum for this. And convert OS defines, e.g. 'SO_EE_CODE_ZEROCOPY_COPIED' to something human readable. Like 
'e_COPY_AVOIDED' and 'e_COPY_DONE'